apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfig.java

```diff
@@ -103,37 +172,41 @@ public String getProperty(String key, String defaultValue) {
     }
 
     // step 4: check properties file from classpath
-    if (value == null && m_resourceProperties != null) {
-      value = m_resourceProperties.getProperty(key);
+    if (value == null) {
+      value = this.getPropertyFromAdditional(key);
     }
 
-    if (value == null && m_configProperties.get() == null && m_warnLogRateLimiter.tryAcquire()) {
-      logger.warn(
-          "Could not load config for namespace {} from Apollo, please check whether the configs are released in Apollo! Return default value now!",
-          m_namespace);
-    }
+    this.tryWarnLog(value);
 
     return value == null ? defaultValue : value;
   }
 
   @Override
   public Set<String> getPropertyNames() {
-    Properties properties = m_configProperties.get();
-    if (properties == null) {
-      return Collections.emptySet();
+    // propertyNames include system property and system env might cause some compatibility issues, though that looks like the correct implementation.
+    Set<String> fromRepository = this.getPropertyNamesFromRepository();
+    Set<String> fromAdditional = this.getPropertyNamesFromAdditional();
+    if (CollectionUtils.isEmpty(fromRepository)) {
+      return fromAdditional;
     }
-
-    return stringPropertyNames(properties);
+    if (CollectionUtils.isEmpty(fromAdditional)) {
+      return fromRepository;
+    }
+    Set<String> propertyNames = Sets
+        .newLinkedHashSetWithExpectedSize(fromRepository.size() + fromAdditional.size());
+    propertyNames.addAll(fromRepository);
+    propertyNames.addAll(fromAdditional);
+    return propertyNames;
   }
 
   @Override
   public ConfigSourceType getSourceType() {
     return m_sourceType;
   }
 
-  private Set<String> stringPropertyNames(Properties properties) {
+  protected Set<String> stringPropertyNames(Properties properties) {
```

Comment:
It seems it could still be private?
```suggestion
  private Set<String> stringPropertyNames(Properties properties) {
```